### ISSUE TITLE
[BUG] cannot import API if in the configuration, requestPolicy is omitted

### ISSUE THREAD

**Issue by OlivierWintquin** [Axway-API-Management-Plus/apimanager-swagger-promote#156]
**Swagger-Promote Version**
1.6.3-SNAPSHOT

**API-Manager and Service-Pack Version**
7.5.3 SP7

**Expected behavior**
Publish the API without request Policy configured
  "outboundProfiles": {
    "_default": {
      "responsePolicy": "GlobalResponsePolicy"
    }
  }

**Actual behavior**
Cannot Deploy API.

2219   ERROR      OutboundProfile| Available request policies: [GlobalRequestPolicy]
2223   ERROR                  App| The policy: '' is not configured in this API-Manager

**Test case sample**
> Please describe the steps you have done to replicate the issue
put the following outboundProfiles configuration in your file

"outboundProfiles": {
  "_default": {
    "responsePolicy": "GlobalResponsePolicy"
  }
}

**Comment by cwiechmann** [comment]
I can't reproduce this issue. I have used the following config:
```json
	"outboundProfiles": {
		"_default": {
			"responsePolicy": "Response policy 1"
		}
	},
```
I'm confused by that message:
`The policy: '' is not configured in this API-Manager`

As it looks like there is an empty Request-Policy config. A further question: Does the API exists and you are trying to update it or is it a complete new one?

**Comment by OlivierWintquin** [comment]
Yes, the API Already exists. Maybe this comes from the "validation" of the existing policy...

**Comment by cwiechmann** [comment]
I could reproduce the issue on my side. The code was not prepared to accept Empty-Strings as to be a policy-name, but that's the way how it's returned from the API-Manager REST-API when configured like in your case.
I added a Test-Case for that use-case and improved the handling.

**Comment by stevebower59** [comment]
I am having a similar issue, but on the export of the API.  I've defined a Request Policy and Reponse Policy in the API-Manager, but they don't seem to get exported to the api-config.json.  
If I want to define them for the import/promotion, I'm having to add the outboundProfiles section to the api-config.json before import.  

Is there a way to force the api-export script to write out the outboundProfiles, and inboundProfiles sections if they have been defined?

**Comment by cwiechmann** [comment]
The issue are related as it is caused by the internal creation of the “Actual API”. This happens for the Import & Export of an API.
Both should be fixed by the last changes I made in the Develop branch.

You can test it yourself building you own Snapshot now or wait until tomorrow.

See here how to build a release:
https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/wiki/7.1-Contribute-to-this-project#compile-your-own-swagger-promote-release

**Comment by cwiechmann** [comment]
@OlivierWintquin, @stevebower59 - The issue should have been fixed and is included in this SNAPSHOT-Version. I highly appreciate your feedback. Thanks!

[apimanager-swagger-promote-1.6.3-SNAPSHOT.zip](https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/files/3647876/apimanager-swagger-promote-1.6.3-SNAPSHOT.zip)

**Comment by stevebower59** [comment]
I exported one of my APIS with the 1.6.3-SNAPSHOT api-export utility, and it still does not write out the outboundProfiles section for the FrontEnd API Outbound Tab (Which had Request and Response Handlers defined) into the generated api-config.json API definition file.
Attaching Screenshot (inside zip), as well as the files generated from the api-export.  
[EquipmentMaster.zip](https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/files/3648463/EquipmentMaster.zip)

**Comment by cwiechmann** [comment]
I'm going to create an integration-test with that API-Definition.

**Comment by cwiechmann** [comment]
The new IntegrationTest: CustomPoliciesTestIt now includes a validation if the OutboundProfile is exported and that all Policies imported before are exported.
Additionally I also made a change to translate the used Tokeninformation- or Inbound-Security-Policy.

**Comment by cwiechmann** [comment]
@stevebower59, @OlivierWintquin 
[apimanager-swagger-promote-1.6.3-SNAPSHOT.zip](https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/files/3651486/apimanager-swagger-promote-1.6.3-SNAPSHOT.zip)

**Comment by stevebower59** [comment]
Hi @cwiechmann,

I tried the new snapshot, and it does not export the OutboundProfiles into the api-config.json - but it now DOES export the trusted certificates for issue #157.  I'll do some more testing on other APIs that I need to export.

**Comment by stevebower59** [comment]
[api-config.zip](https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/files/3653166/api-config.zip)

**Comment by cwiechmann** [comment]
I cannot re-produce the issue you have. I created an Integration-Test: 
[CustomPoliciesTestIt.java](https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/blob/develop/modules/swagger-promote-export/src/test/java/com/axway/apim/export/test/customPolicies/CustomPoliciesTestIt.java)

Which is using the following API-Config file to import an API: 
[custom-policies-issue-156.json](https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/blob/develop/modules/swagger-promote-export/src/test/resources/test/export/files/customPolicies/custom-policies-issue-156.json) <-- This is template, hence policy-names are replaced.
And afterwards the same API is exported. The Integration-Tests is validating, that the Import-Definition equals the Export-Definition. This also includes the Outbound-Profiles.

This test is working fine on all tested versions (>=7.6.2)

**Comment by stevebower59** [comment]
Thanks @cwiechmann.

It looks like I tried to export one of my APIs that did not have an outBoundProfiles defined.  

I tried it on another of my API's that DID have an outBoundProfiles Request and Response Policy, and it does indeed export that section - see attached.

I'll do some more testing on some of my other APIs and let you know if I run into any other issues. 

Thanks for your help!

[api-config-CerbData.zip](https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/files/3653774/api-config-CerbData.zip)
